api/metadata/module.go

```diff
@@ -61,7 +81,62 @@ func (m Module) buildMetadata() primitives.Metadata {
 		Type:      sc.ToCompact(metadata.Runtime),
 	}
 
-	return primitives.NewMetadata(runtimeV14Metadata)
+	return primitives.Metadata{
+		Version: primitives.MetadataVersion14,
+		DataV14: runtimeV14Metadata,
+	}
+}
+
+// MetadataAtVersion returns the metadata of a specific version of the runtime passed as argument.
+// Returns a pointer-size of the SCALE-encoded metadata of the runtime.
+// [Specification](https://spec.polkadot.network/chap-runtime-api#sect-rte-metadata-metadata)
+func (m Module) MetadataAtVersion(dataPtr int32, dataLen int32) int64 {
```

Comment:
This method returns an option of metadata, as some versions might not be supported.